aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -785,16 +791,66 @@ module aptos_framework::fungible_asset {
         amount: u64,
     ): FungibleAsset acquires FungibleStore, DispatchFunctionStore, ConcurrentFungibleBalance {
         withdraw_sanity_check(owner, store, true);
+        withdraw_permission_check(owner, store, amount);
         withdraw_internal(object::object_address(&store), amount)
     }
 
+    public fun withdraw_with_permission<T: key>(
```

Comment:
let's move to the separate PR that adds Permission struct. There we can also test perf of this.